tools/runner/runner.go

```diff
@@ -141,11 +141,15 @@ func (r *Runner) runTest(ctx context.Context, config *grpcv1.LoadTest, reporter
 		status = statusString(config)
 		switch {
 		case loadTest.Status.State.IsTerminated():
-			_, err := r.logSaver.SavePodLogs(ctx, loadTest, outputDir)
+			podToPathMap, err := r.logSaver.SavePodLogs(ctx, loadTest, outputDir)
 			if err != nil {
 				reporter.Error("Could not save pod logs: %s", err)
 			}
 			reporter.AddProperty("name", loadTest.Name)
+			for pod := range podToPathMap {
```

Comment:
Intricate logic like this should not be here. It should be in its own function. You might even define the "podToPathMap" as its own type and make it a method that returns another map or other data structure containing the key-value pairs for properties. Here you would iterate over that and add them to properties.